src/app/backend/integration/metric/sidecar/client.go

```diff
@@ -0,0 +1,254 @@
+// Copyright 2017 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sidecar
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/kubernetes/dashboard/src/app/backend/client"
+	integrationapi "github.com/kubernetes/dashboard/src/app/backend/integration/api"
+	metricapi "github.com/kubernetes/dashboard/src/app/backend/integration/metric/api"
+	"github.com/kubernetes/dashboard/src/app/backend/integration/metric/common"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// Sidecar client implements MetricClient and Integration interfaces.
+type sidecarClient struct {
+	client SidecarRESTClient
+}
+
+// Implement Integration interface.
```

Comment:
TODO?

Reply:
No, the Sidecar code borrows heavily (read: copied, pasted, then modified) from the Heapster metric integration. The Heapster integration has the [same comment](https://github.com/kubernetes/dashboard/blob/master/src/app/backend/integration/metric/heapster/client.go#L39) in it.